src/passport-saml/xml-utils.ts

```diff
@@ -0,0 +1,119 @@
+import * as util from "util";
+import * as xmlCrypto from "xml-crypto";
+import * as xmlenc from "xml-encryption";
+import * as xmldom from "xmldom";
+import * as xml2js from "xml2js";
+import * as xmlbuilder from "xmlbuilder";
+
+type SelectedValue = string | number | boolean | Node;
+
+const selectXPath = <T extends SelectedValue>(
+  guard: (values: SelectedValue[]) => values is T[],
+  node: Node,
+  xpath: string
+): T[] => {
+  const result = xmlCrypto.xpath(node, xpath);
+  if (!guard(result)) {
```

Comment:
This ensure we get the expected type from the xpath lib at runtime. It should not break anything, and if it does, it's probably for the best.